src/services/Forum/HandleForumPost.js

```diff
@@ -0,0 +1,165 @@
+import { Alert } from "react-native";
+import { firebase } from "../Firebase/Config";
+import Warning from "../../components/Forum/Warning";
+import { likeStatus } from "../../constants/Post";
+
+export async function addPost(forumId, post, onSuccess, onError) {
+  const currentUID = firebase.auth().currentUser.uid;
+
+  //Create post
+  await firebase
+    .firestore()
+    .collection("forums")
+    .doc(forumId)
+    .collection("posts")
+    .add({ ...post, uid: currentUID })
+    .then((doc) => {
+      //Then add to user history
+      firebase
+        .firestore()
+        .collection("users")
+        .doc(currentUID)
+        .collection("posts")
+        .add({
+          postId: doc.id,
+          forumId: forumId,
+          timestamp: new Date()
+        });
+      onSuccess();
+    })
+    .catch((e) => onError(e));
+}
+
+export async function deletePost(forumId, postId, onSuccess, onError) {
+  console.log("run");
+  Warning(async () => {
```

Comment:
For testing reasons, I think it would be better if we use the Warning in other files.